doc/sphinx/src/using-closures.rst

```diff
@@ -3,131 +3,362 @@
+
+where :math:`f_\mathrm{tot}` is the total fraction of the total volume being
+considered (in principle, different closure models can be used for different
+sets of materials). To consider the entire volume, :math:`f_\mathrm{tot}` can
+simply be set to one.
+
+The average density, :math:`\overline{\rho}_i`, (i.e. mass per *total* volume)
+for a material in the total volume is
+
+.. math::
+
+  \overline{\rho}_i = \rho_i f_i,
+
+where :math:`\rho_i` is the physical density (i.e. material mass per *material*
+volume). The total density (mass of *participating* materials per total volume)
+is then
+
+.. math::
+
+  \rho = \sum_{i=0}^{N - 1} \overline{\rho}_i = \sum_{i=0}^{N-1} \rho_i f_i
+
+Similarly the energy can be summed in a similar way so that
+
+.. math::
+
+  u = \rho \epsilon = \sum_{i = 0}^{N - 1} \rho_i \epsilon_i 
+  = \sum_{i = 0}^{N - 1} u_i
+
+where :math:`u` is the total internal energy density (internal energy per unit
+volume). Similarly, :math:`u_i` is analagous to :math:`\overline{\rho}_i` in
+that it is the internal energy for a material averaged over the entire control
+volume.
+
+Internally, the closer models in ``singularity-eos`` operate on :math:`f_i`,
+:math:`\overline{\rho}_i`, and :math:`u_i` as well as their total counterparts.
+ This is different than the forms stated at the beginning of this section so
+ that in essence the PTE solver has the form
+
+.. math::
+
+  P_i = F(\epsilon, f_\mathrm{tot}, f_1, ..., f_i, ..., f_{N-1},
+          \rho_1, ..., \rho_i, ..., \rho_{N-1}) \\
+  \rho_i = G(\epsilon, f_\mathrm{tot}, f_1, ..., f_i, ..., f_{N-1},
+          \rho_1, ..., \rho_i, ..., \rho_{N-1}) \\
+  \epsilon_i = H(\epsilon, f_\mathrm{tot}, f_1, ..., f_i, ..., f_{N-1},
+          \rho_1, ..., \rho_i, ..., \rho_{N-1}) \\
+  T_i = J(\epsilon, f_\mathrm{tot}, f_1, ..., f_i, ..., f_{N-1},
+          \rho_1, ..., \rho_i, ..., \rho_{N-1})
+
+
+The important nuance here is the **the volume fractions and densities are both
+inputs and outputs** in the current ``singularity-eos`` formulation of the
+closure models. From physics perspective this can be confusing, but from a code
+perspective this limits the number of variables that need to be passed to the
+PTE solver and provides a convenient way to specify an initial guess for the
+closure state.
+
+.. warning::
+
+  The volume fractions and material densities must be consistent in some way
+
+.. note::
+
+  The mass fraction information is encoded in the specifciation of :math:`f_i`
+  and :math:`\rho_i`. In order to convert to component densities,
+  :math:`\rho_i` from mass fractions, :math:`\mu_i` and total density,
+  :math:`\rho`, volume fractions must be ***assumed*** in some consistent way
+  so that
+
+  .. math::
+
+    \rho_i = \frac{\mu_i \rho}{f_i}
+
+  .. math::
+
+    \sum\limits_{i=0}^{N-1} f_i = f_\mathrm{tot}.
+
+  One such assumption would be to **equipartition** the volume between all
+  materials such that
+
+  .. math::
+
+    f_i = \frac{1}{N}
```

Comment:
This doesn't work for most of the test I have run. It is much more sensitive than that. For the 3-phase problem/test case I made I had to start much more close to the final results, and even the 2-phase problems/test cases I made it would have been too crude if it wasn't for the fact that the final results were close to 50/50.

Please add that the previous time step's volume fractions might be a good guess.
Or, if you have a god sense on the individual densities of the phases, putting these in as a guess and then set the volume fractions accordingly (as I showed in my write up) might be better.

It took me quite a while to find settings that worked.

Reply:
thanks for the feedback! I added a note about using the previous timestep's values to the top of this block and added a discussion about different assumptions that can be made.

A better assumption is just to use the mass fractions, and yet a better assumption is to weight the mass fractions by the material bulk moduli. I added this to the discussion as well.